src/components/pages/home.jsx

```diff
@@ -32,28 +34,32 @@ const HomePage = React.createClass({
     return {inputValue: ""}
   },
   handleClearSearchClicked() {
-    this.setState({inputValue: ""})
-    this.context.router.push("")
+    const query = ""
+    this.setState({inputValue: query})
+    this.context.setQuery(query)
+    this.context.router.push(query)
   },
   handleInputChange(event) {
     this.setState({inputValue: event.target.value})
     this.searchInput.scrollIntoView()
   },
   handleSubmit(event) {
     event.preventDefault()
+    this.context.setQuery(this.state.inputValue)
     this.context.router.push(`?q=${this.state.inputValue}#${searchInputId}`)
```

Comment:
I don't understand if your remark is about line 48 or 49.

On line 48 there is no query string: `this.context.setQuery` sets the *search query* in the global state, accessed through `context` (a React specificity...).